tests/linearGaussian_npse_test.py

```diff
@@ -1,300 +1,281 @@
     )
-    inference.train(
-        resume_training=True, force_first_round_loss=True, training_batch_size=100
+    theta, x = test_case.get_training_data(num_simulations)
+    inference.append_simulations(theta, x)
+    inference, score_estimator = _train_npse(
+        test_case, num_simulations, stop_after_epochs, training_batch_size
     )
-    posterior = inference.build_posterior().set_default_x(x_o)
-    samples = posterior.sample((num_samples,))
-
-    # Compute the c2st and assert it is near chance level of 0.5.
-    check_c2st(samples, target_samples, alg="npse_different_dims_and_resume_training")
-
-
-@pytest.fixture(scope="module", params=["vp", "ve", "subvp"])
-def sde_type(request):
-    """Module-scoped fixture for SDE type."""
-    return request.param
-
+    return inference, score_estimator, test_case
 
-@pytest.fixture(scope="module", params=["gaussian", "uniform", None])
-def prior_type(request):
-    """Module-scoped fixture for prior type."""
-    return request.param
 
-
-@pytest.fixture(scope="module")
-def npse_trained_model(sde_type, prior_type):
-    """Module-scoped fixture that trains a score estimator for NPSE tests."""
-    num_dim = 2
-    num_simulations = 5000
-
-    # likelihood_mean will be likelihood_shift+theta
-    likelihood_shift = -1.0 * ones(num_dim)
-    likelihood_cov = 0.3 * eye(num_dim)
-
-    if prior_type == "gaussian" or (prior_type is None):
-        prior_mean = zeros(num_dim)
-        prior_cov = eye(num_dim)
-        prior = MultivariateNormal(loc=prior_mean, covariance_matrix=prior_cov)
-        prior_npse = prior if prior_type is None else None
-    elif prior_type == "uniform":
-        prior = BoxUniform(-2 * ones(num_dim), 2 * ones(num_dim))
-        prior_npse = prior
-
-    # This check that our method to handle "general" priors works.
-    # i.e. if NPSE does not get a proper passed by the user.
-    inference = NPSE(prior_npse, show_progress_bars=True, sde_type=sde_type)
-
-    theta = prior.sample((num_simulations,))
-    x = linear_gaussian(theta, likelihood_shift, likelihood_cov)
-
-    score_estimator = inference.append_simulations(theta, x).train(
-        stop_after_epochs=200
+@pytest.mark.slow
+@pytest.mark.parametrize("npse_trained_model", training_test_cases_all, indirect=True)
+@pytest.mark.parametrize("sampling_test_case", sampling_test_cases_all)
+def test_c2st(npse_trained_model, sampling_test_case: NpseSamplingTestCase):
+    num_samples = 1_000
+    inference, score_estimator, test_case = npse_trained_model
+
+    x_o = zeros(sampling_test_case.num_trials, test_case.num_dim)
+    posterior = inference.build_posterior(
+        score_estimator, sample_with=sampling_test_case.sampling_method
+    )
+    posterior.set_default_x(x_o)
+    npse_samples = posterior.sample(
+        (num_samples,), iid_method=sampling_test_case.iid_method
+    )
+    check_c2st(
+        npse_samples,
+        test_case.get_target_samples(
```

Comment:
how does the (training) `test_case` know about the current `x_o` that is changing with every `sampling_test_case`? I think it should take `x_o` as an argument such that the target posterior samples are adapted to the current `x_o`.